packages/react-native-gesture-handler/src/v3/hooks/useGesture.ts

```diff
@@ -46,6 +49,54 @@ function shouldHandleTouchEvents(config: Record<string, unknown>) {
   );
 }
 
+const SHARED_VALUE_OFFSET = 1.618;
+
+// This is used to obtain HostFunction that can be executed on the UI thread
+const { updateGestureHandlerConfig } = RNGestureHandlerModule;
+
+function bindSharedValues(config: any, tag: number) {
+  if (Reanimated === undefined) {
+    return;
+  }
+
+  const listenerId = tag + SHARED_VALUE_OFFSET;
+
+  const attachListener = (sharedValue: SharedValue, configKey: string) => {
+    'worklet';
+    sharedValue.addListener(listenerId, (value) => {
+      updateGestureHandlerConfig(tag, { [configKey]: value });
+    });
+  };
+
+  for (const [key, maybeSharedValue] of Object.entries(config)) {
+    if (!Reanimated.isSharedValue(maybeSharedValue)) {
+      continue;
+    }
+
+    config[key] = maybeSharedValue.value;
```

Comment:
Doesn't this break the unbinding? This looks like it overwrites the shared value in the config with its value. If so, isn't `unbindSharedValues` doing effectively nothing?

Reply:
Yes, it does, that's why I had used `Map` before. I've changed it to keep `SharedValue` in config and create copy before we pass it to the native side. Done in [ee63683](https://github.com/software-mansion/react-native-gesture-handler/pull/3658/commits/ee6368303ca4eeb4d9ae28c24e03856b33dd490a)